noether_gui/include/noether_gui/widgets/tool_path_modifiers/offset_modifier_widget.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <noether_gui/widgets.h>
+#include <noether_tpp/core/tool_path_modifier.h>
+
+namespace Ui
+{
+class Vector3dEditor;
+}
+
+namespace noether
+{
+class OffsetModifierWidget : public ToolPathModifierWidget
+{
+  Q_OBJECT
+public:
+  OffsetModifierWidget(QWidget* parent = nullptr);
+  ToolPathModifier::ConstPtr create() const override;
+
+  void configure(const YAML::Node&) override;
+  void save(YAML::Node&) const override;
+
+private:
+  Ui::Vector3dEditor* ui_;
```

Comment:
We'll need to create another `.ui` file for quaternion editing so we can specify the full pose. It should look similar to the Vector3dEditor UI